Mathlib/LinearAlgebra/Matrix/Diagonal.lean

```diff
@@ -76,20 +77,39 @@ theorem range_diagonal [DecidableEq m] (w : m → K) :
 
 end Semifield
 
+end Matrix
+
+namespace LinearMap
+
 section Field
 
 variable {m n : Type _} [Fintype m] [Fintype n] {K : Type u} [Field K]
 
 theorem rank_diagonal [DecidableEq m] [DecidableEq K] (w : m → K) :
-    rank (toLin' (diagonal w)) = Fintype.card { i // w i ≠ 0 } := by
+    LinearMap.rank (toLin' (diagonal w)) = Fintype.card { i // w i ≠ 0 } := by
   have hu : univ ⊆ { i : m | w i = 0 }ᶜ ∪ { i : m | w i = 0 } := by rw [Set.compl_union_self]
   have hd : Disjoint { i : m | w i ≠ 0 } { i : m | w i = 0 } := disjoint_compl_left
   have B₁ := iSup_range_stdBasis_eq_iInf_ker_proj K (fun _ : m => K) hd hu (Set.toFinite _)
   have B₂ := iInfKerProjEquiv K (fun _ ↦ K) hd hu
-  rw [rank, range_diagonal, B₁, ← @rank_fun' K]
+  rw [LinearMap.rank, range_diagonal, B₁, ← @rank_fun' K]
   apply LinearEquiv.rank_eq
   apply B₂
-#align matrix.rank_diagonal Matrix.rank_diagonal
+#align matrix.rank_diagonal LinearMap.rank_diagonal
+
+end Field
+
+end LinearMap
+
+namespace Matrix
+
+section Field
+
+variable {m n : Type _} [Fintype m] [Fintype n] {K : Type u} [Field K]
+/-- The rank of a diagnonal matrix  is the count of non-zero elements on its main diagonal -/
+theorem rank_diagonal [DecidableEq m] [DecidableEq K] (w : m → K) :
```

Comment:
I'd argue this result belongs in `Daya.Matrix.Rank`

Reply:
That will cause a build cycle

Reply:
Done